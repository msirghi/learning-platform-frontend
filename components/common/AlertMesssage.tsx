import React from 'react';
import Alert from '@material-ui/lab/Alert';
import { AlertType } from '../../common/enums';

/**
 * AlertMessage component props.
 */
type Props = {
  /**
   * Message that will be displayed
   */
  message: string;
  /**
   * Alert type. Depending on it component will have special colors.
   */
  type: AlertType;
};

/**
 * Component for displaying the alert messages. Uses Alert component from '@material-ui/lab' library.
 *
 * @version 0.1
 * @author [Sirghi Mihail](https://github.com/msirghi)
 */
export const AlertMessage: React.FC<Props> = ({ message, type }) => {
  return <Alert severity={type}>{message}</Alert>;
};
