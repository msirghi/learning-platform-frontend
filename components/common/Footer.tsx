import React from 'react';
import CssBaseline from '@material-ui/core/CssBaseline';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import Link from '@material-ui/core/Link';

function Copyright() {
  return (
    <Typography variant='body2'>
      {'Copyright Â© '}
      <Link color='inherit' href='https://material-ui.com/'>
        Your Website
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

const useStyles = makeStyles((theme) => ({
  footer: {
    padding: theme.spacing(3, 2),
    marginTop: 'auto',
    backgroundColor: '#202020',
    color: '#fff'
  }
}));

export const SiteFooter = () => {
  const classes = useStyles();
  return (
    <footer className={classes.footer}>
      <Container maxWidth='sm'>
        <Typography variant='body1'>My sticky footer can be found here.</Typography>
        <Copyright />
      </Container>
    </footer>
  );
};
