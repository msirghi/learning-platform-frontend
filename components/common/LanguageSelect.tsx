import React, { useState } from 'react';
import LanguageIcon from '@material-ui/icons/Language';
import IconButton from '@material-ui/core/IconButton';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Menu from '@material-ui/core/Menu';
import { useTranslation } from '../../i18n';
import { useDispatch } from 'react-redux';
import { setInterfaceLocale } from '../../redux/actions/preference/preferenceAction';

const languageChoice = [
  {
    key: 'en',
    value: 'English'
  },
  {
    key: 'ru',
    value: 'Русский'
  }
];

export const LanguageSelect: React.FC = () => {
  const [notificationMenuOpen, setNotificationMenuOpen] = useState<null | HTMLElement>(null);
  const { i18n } = useTranslation();
  const dispatch = useDispatch();

  const handleChange = (val: string) => {
    dispatch(setInterfaceLocale(val));
    i18n.changeLanguage(val);
    handleClose();
  };

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setNotificationMenuOpen(event.currentTarget);
  };

  const handleClose = () => {
    setNotificationMenuOpen(null);
  };

  return (
    <>
      <IconButton onClick={handleClick}>
        <LanguageIcon color='primary' />
      </IconButton>
      <Menu
        anchorEl={notificationMenuOpen}
        keepMounted
        open={Boolean(notificationMenuOpen)}
        onClose={handleClose}
      >
        <List>
          {languageChoice.map((item) => {
            return (
              <ListItem key={item.key} button onClick={() => handleChange(item.key)}>
                {item.value}
              </ListItem>
            );
          })}
        </List>
      </Menu>
    </>
  );
};
