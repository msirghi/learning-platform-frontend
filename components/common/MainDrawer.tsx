import React, { useEffect, useState } from "react";
import Drawer from "@material-ui/core/Drawer";
import IconButton from "@material-ui/core/IconButton";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import { makeStyles } from "@material-ui/core/styles";
import clsx from "clsx";
import { menuItems } from "./utils/menuItems";
import styles from "../../styles/modules/Shared.module.scss";
import { Hidden } from "@material-ui/core";

const drawerWidth = 240;

const useStyles = makeStyles((theme) => ({
  drawer: {
    width: drawerWidth,
  },
  drawerOpen: {
    background: theme.palette.primary.main,
    width: drawerWidth,
    transition: theme.transitions.create("width", {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  drawerClose: {
    background: theme.palette.primary.main,
    transition: theme.transitions.create("width", {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    overflowX: "hidden",
    width: theme.spacing(8) + 1,
    [theme.breakpoints.up("sm")]: {
      width: 110,
    },
  },
  toolbar: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    padding: theme.spacing(0, 1),
    ...theme.mixins.toolbar,
  },
  drawerPaper: {
    width: drawerWidth,
    background: theme.palette.primary.main,
  },
}));

type Props = {
  handleDrawerClose: () => void;
  open: boolean;
  mobileOpen: boolean;
};

export const MainDrawer: React.FC<Props> = ({ handleDrawerClose, open, mobileOpen }) => {
  const classes = useStyles();
  const [activeTab, setActiveTab] = useState<string | null>(null);

  useEffect(() => {
    /* istanbul ignore next */
    if (window) {
      setActiveTab(window.location.href);
    }
  }, []);

  if (!activeTab) {
    return <div />;
  }

  const drawerContent = () => {
    return (
      <>
        <div className={classes.toolbar}>
          <IconButton onClick={handleDrawerClose}>
            <img src='/images/register-logo-148.png' height={72} width={72} />
          </IconButton>
        </div>
        <List className={styles.mainMenuListContainer}>
          {menuItems.map((item) => {
            const isActiveTab = activeTab.includes(item.link);
            return (
              <div
                className={`${styles.mainMenuColor} ${isActiveTab && styles.mainMenuActive} ${
                  item.className
                }`}
                key={item.label}
              >
                <ListItem button>
                  <ListItemIcon className={styles.mainMenuIconContainer}>
                    <item.icon
                      className={`${isActiveTab ? styles.mainMenuActiveIcon : styles.mainMenuIcon}`}
                    />
                  </ListItemIcon>
                  {(open || mobileOpen) && (
                    <ListItemText
                      className={isActiveTab ? styles.mainMenuActiveText : ""}
                      primary={item.label}
                    />
                  )}
                </ListItem>
              </div>
            );
          })}
        </List>
      </>
    );
  };

  return (
    <>
      <Drawer
        variant='temporary'
        open={mobileOpen}
        onClose={handleDrawerClose}
        classes={{
          paper: classes.drawerPaper,
        }}
        ModalProps={{
          keepMounted: true,
        }}
      >
        {drawerContent()}
      </Drawer>
      <Hidden xsDown implementation='css'>
        <Drawer
          classes={{
            paper: clsx({
              [classes.drawerOpen]: open,
              [classes.drawerClose]: !open,
            }),
          }}
          className={clsx(classes.drawer, {
            [classes.drawerOpen]: open,
            [classes.drawerClose]: !open,
          })}
          variant='permanent'
          open={open}
        >
          {drawerContent()}
        </Drawer>
      </Hidden>
    </>
  );
};
