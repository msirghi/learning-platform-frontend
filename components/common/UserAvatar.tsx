import React from 'react';

type Props = {
  img?: string;
};

const styles = { height: 35, width: 35, background: '#bbb', borderRadius: '50%' };

export const UserAvatar: React.FC<Props> = ({ img }) => {
  if (img) {
    return <img style={styles} src={img} />;
  }

  return <span style={styles} />;
};
