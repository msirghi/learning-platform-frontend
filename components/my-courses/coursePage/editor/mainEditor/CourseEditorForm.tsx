import { TextField, Button } from '@material-ui/core';
import React from 'react';
import styles from '../CourseEditor.module.scss';
import { DatePicker, TimePicker } from '@material-ui/pickers';
import { useWindowSize } from '../../../../common/hooks/useWindowResize';
import { useTranslation } from '../../../../../i18n';
import { LessonCreationFormValues } from '../../../../../common/types';

type Props = {
  onSubmit: () => void;
  onChange: (name: keyof LessonCreationFormValues, value: string | Date) => void;
  values: LessonCreationFormValues;
};

export const CourseEditorForm: React.FC<Props> = ({ onSubmit, onChange, values }) => {
  const [width] = useWindowSize();
  const { t } = useTranslation();
  const { startDate, endDate, subject, description, startTime, endTime } = values;

  return (
    <div>
      <div className={styles.sectionTitle}>{t('courses:informationLabel')}</div>
      <div>
        <TextField
          onChange={(e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) =>
            onChange('subject', e.target.value)
          }
          inputProps={{
            'data-testid': 'subject-field'
          }}
          value={subject}
          name='subject'
          helperText={t('courses:editor.subjectHelperText')}
          fullWidth
          variant='outlined'
          label={t('courses:editor.subjectLabel')}
          required
        />
      </div>

      <div className={styles.dateTimeRow}>
        <DatePicker
          inputProps={{
            'data-testid': 'start-date-field'
          }}
          value={startDate}
          onChange={(date) => onChange('startDate', date)}
          name='startDate'
          fullWidth={width < 1200}
          helperText={t('courses:editor.startDateLabel')}
          inputVariant='outlined'
        />
        <TimePicker
          inputProps={{
            'data-testid': 'start-time-field'
          }}
          value={startTime}
          onChange={(time) => onChange('startTime', time)}
          name='startTime'
          fullWidth={width < 1200}
          helperText={t('courses:editor.startTimeLabel')}
          className={styles.timepicker}
          inputVariant='outlined'
        />
      </div>

      <div className={styles.dateTimeRow}>
        <DatePicker
          inputProps={{
            'data-testid': 'end-date-field'
          }}
          onChange={(date) => onChange('endDate', date)}
          value={endDate}
          name='endDate'
          fullWidth={width < 1200}
          inputVariant='outlined'
          helperText={t('courses:editor.endDateLabel')}
        />
        <TimePicker
          inputProps={{
            'data-testid': 'end-time-field'
          }}
          value={endTime}
          onChange={(time) => onChange('endTime', time)}
          name='endTime'
          fullWidth={width < 1200}
          inputVariant='outlined'
          helperText={t('courses:editor.endTimeLabel')}
          className={styles.timepicker}
        />
      </div>

      <div>
        <TextField
          onChange={(e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) =>
            onChange('description', e.target.value)
          }
          inputProps={{
            'data-testid': 'description-field'
          }}
          name='description'
          helperText={t('courses:editor.descriptionHelperText')}
          fullWidth
          variant='outlined'
          rows={4}
          multiline
          value={description}
          label={t('courses:editor.descriptionLabel')}
        />
      </div>
      {width > 768 && (
        <Button
          type='submit'
          data-testid='course-submit'
          fullWidth
          className={styles.addButton}
          onClick={onSubmit}
          color='primary'
          variant='contained'
        >
          {t('courses:submitButtonLabel')}
        </Button>
      )}
    </div>
  );
};
