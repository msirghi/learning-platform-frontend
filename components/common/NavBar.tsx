import React, { useEffect, useState } from 'react';
import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Toolbar from '@material-ui/core/Toolbar';
import clsx from 'clsx';
import Typography from '@material-ui/core/Typography';
import NotificationsIcon from '@material-ui/icons/Notifications';
import styles from '../../styles/modules/Shared.module.scss';
import Divider from '@material-ui/core/Divider';
import { UserAvatar } from './UserAvatar';
import Badge from '@material-ui/core/Badge';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import { LanguageSelect } from './LanguageSelect';
import { useTranslation } from '../../i18n';

type Props = {
  classes: { [key: string]: string };
  handleDrawerOpen: () => void;
  handleDrawerClose: () => void;
  open: boolean;
};

export const NavBar: React.FC<Props> = ({ classes, handleDrawerOpen, open, handleDrawerClose }) => {
  const [activeTab, setActiveTab] = useState<string | undefined>();
  const { t, i18n } = useTranslation();

  useEffect(() => {
    if (global.window) {
      console.log('global.window.location :>> ', global.window.location.pathname);
      const href = global.window.location.pathname.replaceAll('/', '');
      setActiveTab(t(`common:headerTabs.${href}`));
    }
  }, [global.window, i18n]);

  return (
    <AppBar
      position='fixed'
      className={clsx(classes.appBar, {
        [classes.appBarShift]: open
      })}
    >
      <Toolbar>
        <IconButton
          color='inherit'
          aria-label='open drawer'
          onClick={handleDrawerOpen}
          edge='start'
          className={clsx(classes.menuButton, {
            [classes.hide]: open
          })}
        >
          <MenuIcon />
        </IconButton>

        <IconButton
          color='inherit'
          aria-label='close drawer'
          onClick={handleDrawerClose}
          edge='start'
          className={clsx(classes.menuButton, {
            [classes.hide]: !open
          })}
        >
          <ChevronLeftIcon />
        </IconButton>
        <Typography variant='h6' noWrap>
          {activeTab}
        </Typography>
        <div className={styles.navBarRightIconsContainer}>
          <LanguageSelect />
          <IconButton>
            <Badge badgeContent={4} color='primary'>
              <NotificationsIcon color={'primary'} />
            </Badge>
          </IconButton>
          <IconButton>
            <UserAvatar />
          </IconButton>
        </div>
      </Toolbar>
      <div className={styles.navBarDividerContainer}>
        <Divider light />
      </div>
    </AppBar>
  );
};
