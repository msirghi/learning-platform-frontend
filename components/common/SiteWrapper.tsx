import React, { useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { CssBaseline } from '@material-ui/core';
import { NavBar } from './NavBar';
import { MainDrawer } from './MainDrawer';
import { SiteFooter } from './Footer';
import { useWindowSize } from './hooks/useWindowResize';

const drawerWidth = 240;

const useStyles = makeStyles((theme) => ({
  wrapper: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh'
  },
  root: {
    display: 'flex'
  },
  main: {
    marginTop: theme.spacing(8),
    marginBottom: theme.spacing(2)
  },
  appBar: {
    backgroundColor: '#fafafa',
    color: '#000',
    boxShadow: 'none',
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth / 2}px)`
    }
  },
  appBarShift: {
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen
    })
  },
  menuButton: {
    marginRight: 36
  },
  hide: {
    display: 'none'
  },
  toolbar: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: theme.spacing(0, 1),
    ...theme.mixins.toolbar
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3)
  }
}));

export const SiteWrapper: React.FC = ({ children }) => {
  const classes = useStyles();
  const [desktopOpen, setDesktopOpen] = React.useState(false);
  const [mobileOpen, setMobileOpen] = React.useState(false);
  const [width] = useWindowSize();

  useEffect(() => {
    if (width < 768 && desktopOpen) {
      setDesktopOpen(false);
      return;
    }

    if(width > 768 && mobileOpen) {
      setMobileOpen(false);
    }
  }, [width]);

  const handleDrawerOpen = () => {
    if (width > 768) {
      setDesktopOpen(true);
      return;
    }
    setMobileOpen(true);
  };

  const handleDrawerClose = () => {
    if (width > 768) {
      setDesktopOpen(false);
      return;
    }
    setMobileOpen(false);
  };

  return (
    <div className={classes.wrapper}>
      <div className={classes.root}>
        <CssBaseline />
        <NavBar
          classes={classes}
          handleDrawerClose={handleDrawerClose}
          handleDrawerOpen={handleDrawerOpen}
          open={desktopOpen}
        />
        <MainDrawer
          handleDrawerOpen={handleDrawerOpen}
          mobileOpen={mobileOpen}
          handleDrawerClose={handleDrawerClose}
          open={desktopOpen}
        />
        <main className={classes.content}>
          <div className={classes.toolbar} />
          {children}
        </main>
      </div>
      <SiteFooter />
    </div>
  );
};
