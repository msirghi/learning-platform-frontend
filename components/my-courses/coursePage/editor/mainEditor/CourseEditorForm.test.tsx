import { Button, TextField } from '@material-ui/core';
import { DatePicker, TimePicker } from '@material-ui/pickers';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { LessonCreationFormValues } from '../../../../../common/types';
import { CourseEditorForm } from './CourseEditorForm';
import DateFnsUtils from '@date-io/date-fns';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';

describe('CourseEditorForm component', () => {
  const defaultProps = {
    onSubmit: jest.fn(),
    onChange: jest.fn((d) => d),
    values: {
      subject: 'Subject',
      endDate: new Date(),
      endTime: new Date(),
      startDate: new Date(),
      startTime: new Date(),
      description: 'description'
    } as LessonCreationFormValues
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot if all the props are provided', () => {
    const wrapper = shallow(<CourseEditorForm {...defaultProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should fire onChange prop function on subject field change', () => {
    const wrapper = shallow(<CourseEditorForm {...defaultProps} />);
    const subjectInput = wrapper.find(TextField).at(0);

    subjectInput.simulate('change', { target: { value: 'New value' } });
    expect(defaultProps.onChange).toBeCalled();
  });

  it('should fire onChange prop function on description field change', () => {
    const wrapper = shallow(<CourseEditorForm {...defaultProps} />);
    const subjectInput = wrapper.find(TextField).at(1);

    subjectInput.simulate('change', { target: { value: 'New value' } });
    expect(defaultProps.onChange).toBeCalled();
  });

  it('should fire onChange prop function on start date field change', () => {
    const wrapper = shallow(<CourseEditorForm {...defaultProps} />);
    const picker = wrapper.find(DatePicker).at(0);

    picker.simulate('change', { target: { value: new Date() } });
    expect(picker.props().helperText).toEqual('courses:editor.startDateLabel');
    expect(defaultProps.onChange).toBeCalledTimes(1);
  });

  it('should fire onChange prop function on end date field change', () => {
    const wrapper = shallow(<CourseEditorForm {...defaultProps} />);
    const picker = wrapper.find(DatePicker).at(1);

    picker.simulate('change', { target: { value: new Date() } });
    expect(picker.props().helperText).toEqual('courses:editor.endDateLabel');
    expect(defaultProps.onChange).toBeCalledTimes(1);
  });

  it('should fire onChange prop function on end time field change', () => {
    const wrapper = shallow(<CourseEditorForm {...defaultProps} />);
    const picker = wrapper.find(TimePicker).at(1);

    picker.simulate('change', { target: { value: new Date() } });
    expect(picker.props().helperText).toEqual('courses:editor.endTimeLabel');
    expect(defaultProps.onChange).toBeCalledTimes(1);
  });

  it('should fire onChange prop function on start time field change', () => {
    const wrapper = shallow(<CourseEditorForm {...defaultProps} />);
    const picker = wrapper.find(TimePicker).at(0);

    picker.simulate('change', { target: { value: new Date() } });
    expect(picker.props().helperText).toEqual('courses:editor.startTimeLabel');
    expect(defaultProps.onChange).toBeCalledTimes(1);
  });

  it('should not render submit button on mobile view', () => {
    const initialWidth = global.innerWidth;
    global.innerWidth = 500;
    global.dispatchEvent(new Event('resize'));

    const wrapper = mount(
      <MuiPickersUtilsProvider utils={DateFnsUtils}>
        <CourseEditorForm {...defaultProps} />
      </MuiPickersUtilsProvider>
    );
    const button = wrapper.find(Button);
    expect(button).toHaveLength(0);

    global.innerWidth = initialWidth;
    global.dispatchEvent(new Event('resize'));
  });

  it('should render submit button on desktop', () => {
    const wrapper = mount(
      <MuiPickersUtilsProvider utils={DateFnsUtils}>
        <CourseEditorForm {...defaultProps} />
      </MuiPickersUtilsProvider>
    );
    const button = wrapper.find(Button);
    expect(button).toHaveLength(1);
  });

  it('should fire onSubmit prop function clicking the submit button', () => {
    const initialWidth = global.innerWidth;
    global.innerWidth = 1024;
    global.dispatchEvent(new Event('resize'));

    const wrapper = mount(
      <MuiPickersUtilsProvider utils={DateFnsUtils}>
        <CourseEditorForm {...defaultProps} />
      </MuiPickersUtilsProvider>
    );
    const button = wrapper.find(Button);

    expect(button).toHaveLength(1);
    button.simulate('click');
    expect(defaultProps.onSubmit).toBeCalled();

    global.innerWidth = initialWidth;
    global.dispatchEvent(new Event('resize'));
  });
});
