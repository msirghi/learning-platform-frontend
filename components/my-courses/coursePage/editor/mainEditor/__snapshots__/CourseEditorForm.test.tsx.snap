// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseEditorForm component should match the snapshot if all the props are provided 1`] = `
<div>
  <div
    className="sectionTitle"
  >
    courses:informationLabel
  </div>
  <div>
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      helperText="courses:editor.subjectHelperText"
      label="courses:editor.subjectLabel"
      name="subject"
      onChange={[Function]}
      required={true}
      value="Subject"
      variant="outlined"
    />
  </div>
  <div
    className="dateTimeRow"
  >
    <PickerWithState
      allowKeyboardControl={true}
      fullWidth={true}
      helperText="courses:editor.startDateLabel"
      inputVariant="outlined"
      invalidDateMessage="Invalid Date Format"
      maxDate={2100-01-01T00:00:00.000Z}
      maxDateMessage="Date should not be after maximal date"
      minDate={1900-01-01T00:00:00.000Z}
      minDateMessage="Date should not be before minimal date"
      name="startDate"
      onChange={[Function]}
      openTo="date"
      value={2000-11-22T00:00:00.000Z}
      views={
        Array [
          "year",
          "date",
        ]
      }
    />
    <PickerWithState
      ampm={true}
      className="timepicker"
      fullWidth={true}
      helperText="courses:editor.startTimeLabel"
      inputVariant="outlined"
      invalidDateMessage="Invalid Time Format"
      name="startTime"
      onChange={[Function]}
      openTo="hours"
      value={2000-11-22T00:00:00.000Z}
      views={
        Array [
          "hours",
          "minutes",
        ]
      }
    />
  </div>
  <div
    className="dateTimeRow"
  >
    <PickerWithState
      allowKeyboardControl={true}
      fullWidth={true}
      helperText="courses:editor.endDateLabel"
      inputVariant="outlined"
      invalidDateMessage="Invalid Date Format"
      maxDate={2100-01-01T00:00:00.000Z}
      maxDateMessage="Date should not be after maximal date"
      minDate={1900-01-01T00:00:00.000Z}
      minDateMessage="Date should not be before minimal date"
      name="endDate"
      onChange={[Function]}
      openTo="date"
      value={2000-11-22T00:00:00.000Z}
      views={
        Array [
          "year",
          "date",
        ]
      }
    />
    <PickerWithState
      ampm={true}
      className="timepicker"
      fullWidth={true}
      helperText="courses:editor.endTimeLabel"
      inputVariant="outlined"
      invalidDateMessage="Invalid Time Format"
      name="endTime"
      onChange={[Function]}
      openTo="hours"
      value={2000-11-22T00:00:00.000Z}
      views={
        Array [
          "hours",
          "minutes",
        ]
      }
    />
  </div>
  <div>
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      helperText="courses:editor.descriptionHelperText"
      label="courses:editor.descriptionLabel"
      multiline={true}
      name="description"
      onChange={[Function]}
      rows={4}
      value="description"
      variant="outlined"
    />
  </div>
</div>
`;
